src/libpriv/rpmostree-scripts.cxx

```diff
@@ -365,17 +366,14 @@ run_script_in_bwrap_container (int rootfs_fd,
   const char *id = NULL;
   int fd = -1;
 
-  /* TODO - Create a pipe and send this to bwrap so it's inside the
-   * tmpfs.  Note the +1 on the path to skip the leading /.
-   */
-  if (!glnx_file_replace_contents_at (rootfs_fd, postscript_path_host,
-                                      (guint8*)script, -1,
-                                      GLNX_FILE_REPLACE_NODATASYNC,
-                                      NULL, error))
-    {
-      g_prefix_error (error, "Writing script to %s: ", postscript_path_host);
-      goto out;
-    }
+  glnx_fd_close int script_memfd = memfd_create (pkg_script, MFD_CLOEXEC | MFD_ALLOW_SEALING);
```

Comment:
Unless you are in a hurry, this data-to-sealed-memfd looks like a useful primitive to have, and the `nix` crate should already have all the required helpers for memfd handling.

Reply:
There's also https://crates.io/crates/memfd right?  But...mmm, yeah we could factor that out into a helper in utils.